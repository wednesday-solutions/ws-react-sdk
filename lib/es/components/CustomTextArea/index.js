import{_ as e,a as r}from"../_rollupPluginBabelHelpers-c2a46d82.js";import"antd/lib/input/style";import n from"antd/lib/input";import"../_rollup-plugin-inject-process-env-1259e43d.js";import t from"react";import{c as o}from"../colors-93a064d8.js";import a from"styled-components";import{m as i}from"../media-ee9cbc51.js";var p,{TextArea:m}=n,l=a(m)(p||(p=e(["\n  height: 7rem;\n  ","\n  ","\n  &.ant-input, &.ant-input-affix-wrapper {\n    padding-left: 1rem;\n    padding-top: 1rem;\n    background-color: ",";\n    border: 1px solid ",";\n    width: 100%;\n    border-radius: 8px;\n    &.ant-form-item-label {\n      font-size: 20px;\n    }\n    &::placeholder {\n      color: ",";\n    }\n  }\n"])),i.desktop.max("\n    height: 3rem;\n  "),i.largeMobile.max("\n    height: 2.5rem;\n  "),o.white,o.gray,o.gray);function d(e){return t.createElement(l,r({"data-testid":"custom-textarea"},e))}export default d;
