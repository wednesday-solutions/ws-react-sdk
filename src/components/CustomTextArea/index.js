/**
 *
 * CustomTextArea
 *
 */

import React from 'react';
import { Input } from 'antd';
import colors from '@themes/colors';
import styled from 'styled-components';
import { media } from '@themes';

const { TextArea } = Input;
const CT = styled(TextArea)`
  height: 7rem;
  ${media.desktop.max(`
    height: 3rem;
  `)}
  ${media.largeMobile.max(`
    height: 2.5rem;
  `)}
  &.ant-input, &.ant-input-affix-wrapper {
    padding-left: 1rem;
    padding-top: 1rem;
    background-color: ${colors.white};
    border: 1px solid ${colors.gray};
    width: 100%;
    border-radius: 8px;
    &.ant-form-item-label {
      font-size: 20px;
    }
    &::placeholder {
      color: ${colors.gray};
    }
  }
`;

function CustomTextArea(props) {
  return <CT data-testid={'custom-textarea'} {...props} />;
}

export default CustomTextArea;
