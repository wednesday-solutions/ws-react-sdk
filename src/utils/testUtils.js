import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { DEFAULT_LOCALE, translationMessages } from '@src/i18n';
import IntlGlobalProvider from '@components/IntlGlobalProvider';

export const renderWithIntl = children => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );
};

export const getComponentStyles = (Component, props = {}) => {
  renderWithIntl(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);
  // eslint-disable-next-line no-underscore-dangle
  return window.getComputedStyle(componentRoots[0])._values;
};

export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export default {
  timeout,
  renderWithIntl,
  getComponentStyles,
  apiResponseGenerator
};
