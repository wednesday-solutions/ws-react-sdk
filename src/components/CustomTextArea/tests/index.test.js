/**
 *
 * Tests for CustomTextArea
 *
 */

import React from 'react';
import { renderWithIntl } from '@utils/testUtils';
import CustomTextArea from '../index';

describe('<CustomTextArea />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<CustomTextArea />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 CustomTextArea component', () => {
    const { getAllByTestId } = renderWithIntl(<CustomTextArea />);
    expect(getAllByTestId('custom-textarea').length).toBe(1);
  });
});
