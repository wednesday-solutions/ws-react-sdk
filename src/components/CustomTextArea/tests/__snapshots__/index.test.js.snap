// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomTextArea /> should render and match the snapshot 1`] = `
<body>
  <div>
    <textarea
      class="ant-input sc-bdnxRM cRWUvh"
      data-testid="custom-textarea"
    />
  </div>
</body>
`;
